File: 74-search-a-2d-matrix/search-a-2d-matrix.py
class Solution(object):
    def searchMatrix(self, matrix, target):
        """
        :type matrix: List[List[int]]
        :type target: int
        :rtype: bool
        """
        n=len(matrix[0])
        m=len(matrix)
        left,right=0,n*m-1
        while left<=right:
            mid=(left+right)//2
            mid_value=matrix[mid//n][mid%n]
            if mid_value==target:
                return True
            elif mid_value>target:
                right=mid-1
            else:
                left=mid+1
        return False
        

File: 74-search-a-2d-matrix/search-a-2d-matrix.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int n=matrix.length;
        int m=matrix[0].length;
        int low=0;
        int hight=m*n-1;
        while(low<=hight){
            int mid=(hight+low)/2;
            if(matrix[mid/m][mid%m]==target){
                return true;
            }
            else if(matrix[mid/m][mid%m]<target){
                low=mid+1;
            }else{
                hight=mid-1;
            }
        }
        return false;
    }
}